src/conic_form.jl

```diff
@@ -0,0 +1,274 @@
+mutable struct ConicForm{T, AT, VT, C} <: MOI.ModelLike
+    num_rows::Vector{Int}
+    dimension::Dict{Int, Int}
+    sense::MOI.OptimizationSense
+    objective_constant::T # The objective
+    A::Union{Nothing, AT} # The constraints
+    b::VT          # `b - Ax in cones`
+    c::VT          # `sense c'x + objective_constant`
+    cone_types::C
+
+    function ConicForm{T, AT, VT}(cone_types) where {T, AT, VT}
+        model = new{T, AT, VT, typeof(cone_types)}()
+        model.cone_types = cone_types
+        model.num_rows = zeros(Int, length(cone_types))
+        model.dimension = Dict{Int, Int}()
+        model.A = nothing
+        return model
+    end
+end
+
+# The ordering of CONES matches SCS
+const CONES = Dict(
+    MOI.Zeros => 1,
+    MOI.Nonnegatives => 2,
+    MOI.SecondOrderCone => 3,
+    MOI.PositiveSemidefiniteConeTriangle => 4,
+    MOI.ExponentialCone => 5, 
+    MOI.DualExponentialCone => 6
+)
+
+cons_offset(conic::MOI.Zeros) = conic.dimension
+cons_offset(conic::MOI.Nonnegatives) = conic.dimension
+cons_offset(conic::MOI.SecondOrderCone) = conic.dimension
+cons_offset(conic::MOI.PositiveSemidefiniteConeTriangle) = Int64((conic.side_dimension*(conic.side_dimension+1))/2)
+
+# function restructure_arrays(_s::Array{T}, _y::Array{T}, cones::Array{<: MOI.AbstractVectorSet}) where {T}
+#     i=0
+#     s = Array{T}[]
+#     y = Array{T}[]
+#     for conic in cones
+#         offset = cons_offset(conic)
+#         push!(s, _s[i.+(1:offset)])
+#         push!(y, _y[i.+(1:offset)])
+#         i += offset
+#     end
+#     return s, y
+# end
+
+function get_conic_form(::Type{T}, model::M, con_idx) where {T, M <: MOI.AbstractOptimizer}
+    # reorder constraints
+    cis = sort(
+        con_idx, 
+        by = x->CONES[typeof(MOI.get(model, MOI.ConstraintSet(), x))]
```

Comment:
You can use, `_set_type(::MOI.ConstraintSet{F,S}) = S` instead of getting a set just to get its type

Reply:
thanks. added!